src/Type/Php/PregSplitDynamicReturnTypeExtension.php

```diff
@@ -36,17 +47,131 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
+			foreach ($flagType->getConstantScalarValues() as $flag) {
+				if (!is_int($flag)) {
+					return new ErrorType();
+				}
+				$flags[] = $flag;
+			}
+		}
+
+		if (count($patternConstantTypes) === 0 || count($subjectConstantTypes) === 0) {
+			$returnNonEmptyStrings = $flagArg !== null && $this->bitwiseFlagAnalyser->bitwiseOrContainsConstant($flagArg->value, $scope, 'PREG_SPLIT_NO_EMPTY')->yes();
+			if ($returnNonEmptyStrings) {
+				$returnStringType = TypeCombinator::intersect(
+					new StringType(),
+					new AccessoryNonEmptyStringType(),
+				);
+			} else {
+				$returnStringType = new StringType();
+			}
 
-		if ($flagsArg !== null && $this->bitwiseFlagAnalyser->bitwiseOrContainsConstant($flagsArg->value, $scope, 'PREG_SPLIT_OFFSET_CAPTURE')->yes()) {
-			$type = new ArrayType(
-				new IntegerType(),
-				new ConstantArrayType([new ConstantIntegerType(0), new ConstantIntegerType(1)], [new StringType(), IntegerRangeType::fromInterval(0, null)], [2], [], TrinaryLogic::createYes()),
+			$capturedArrayType = new ConstantArrayType(
+				[new ConstantIntegerType(0), new ConstantIntegerType(1)],
+				[$returnStringType, IntegerRangeType::fromInterval(0, null)],
+				[2],
+				[],
+				TrinaryLogic::createYes(),
 			);
-			return TypeCombinator::union(TypeCombinator::intersect($type, new AccessoryArrayListType()), new ConstantBooleanType(false));
+
+			$returnInternalValueType = $returnStringType;
+			if ($flagArg !== null) {
+				$flagState = $this->bitwiseFlagAnalyser->bitwiseOrContainsConstant($flagArg->value, $scope, 'PREG_SPLIT_OFFSET_CAPTURE');
+				if ($flagState->yes()) {
+					$capturedArrayListType = TypeCombinator::intersect(
+						new ArrayType(new IntegerType(), $capturedArrayType),
+						new AccessoryArrayListType(),
+					);
+
+					if ($subjectType->isNonEmptyString()->yes()) {
+						$capturedArrayListType = TypeCombinator::intersect($capturedArrayListType, new NonEmptyArrayType());
+					}
+
+					return TypeCombinator::union($capturedArrayListType, new ConstantBooleanType(false));
+				}
+				if ($flagState->maybe()) {
+					$returnInternalValueType = TypeCombinator::union(new StringType(), $capturedArrayType);
+				}
+			}
+
+			$returnListType = TypeCombinator::intersect(new ArrayType(new MixedType(), $returnInternalValueType), new AccessoryArrayListType());
+			if ($subjectType->isNonEmptyString()->yes()) {
+				$returnListType = TypeCombinator::intersect(
+					$returnListType,
+					new NonEmptyArrayType(),
+				);
+			}
+
+			return TypeCombinator::union($returnListType, new ConstantBooleanType(false));
+		}
+
+		$resultTypes = [];
+		foreach ($patternConstantTypes as $patternConstantType) {
+			foreach ($subjectConstantTypes as $subjectConstantType) {
+				foreach ($limits as $limit) {
+					foreach ($flags as $flag) {
+						$result = @preg_split($patternConstantType->getValue(), $subjectConstantType->getValue(), $limit, $flag);
+						if ($result === false) {
+							continue;
```

Comment:
if one of the static analysis time values make `preg_split` return `false` we should give-up instead of ignoring this fact